src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+
+    @Override
+    public Fruit getFruit(Fruit fruit) {
+        Map.Entry<Fruit, Integer> entry = findEntry(fruit);
+        return entry != null ? entry.getKey() : addNewFruit(fruit);
+    }
+
+    @Override
+    public Integer getQuantityFromFruit(Fruit fruit) {
+        Map.Entry<Fruit, Integer> entry = findEntry(fruit);
+        if (entry != null) {
+            return entry.getValue();
+        }
+        throw new RuntimeException(fruit + " does not exist");
```

Comment:
Why RuntimeException??? Return 0 if we do not have it.
```suggestion
        return Storage.getFruits().getOrDefault(fruit, 0)
```